File: mirea/cources/2026-2027/aii-august/06/contest/C.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
void solve() {
    ll x0, y0, r;
    cin >> x0 >> y0 >> r;
    // переносим окружность в (0, 0), т.е. просто игнорируем координаты центра
    // далее идём вдоль левой полуплоскости и для каждого x находим границы столбца
    // [yMin+1, yMax-1], который целиком влезает в окружность. Прибавляем его высоту.
    // Используем два указателя, чтобы эффективно переходить от "x" к "(x-1)".
    ll yMax = 0, yMin = 0, answ = 0;
    for (ll x = -r; x < 0; x++) {
        while (x*x+yMax*yMax <= r*r) yMax++;
        while (x*x+yMin*yMin <= r*r) yMin--;
        answ += max<ll>(0, yMax - yMin - 2);
    }
    cout << answ * 2 << "\n";
}
main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int tt; cin >> tt;
    while (tt --> 0) solve();
}

File: mirea/cources/2026-2027/aii-august/06/contest/C.py
def solve():
    x0, y0, r = map(int, input().split())
    # переносим окружность в (0, 0), то есть просто игнорируем координаты центра
    # далее идём вдоль левой полуплоскости и для каждого x находим границы столбца
    # [yMin+1, yMax-1], который целиком влезает в окружность. Прибавляем его высоту.
    # используем два указателя, чтобы эффективно переходить от x к (x-1).
    yMax, yMin, answ = 0, 0, 0
    for x in range(-r, 0):
        while x * x + yMax * yMax <= r * r:
            yMax += 1
        while x * x + yMin * yMin <= r * r:
            yMin -= 1
        answ += max(0, yMax - yMin - 2)
    print(answ * 2)
tt = int(input())
for _ in range(tt):
    solve()